experiments/AMIP/user_io/leaderboard/data_sources.jl

```diff
@@ -0,0 +1,111 @@
+"""
+
+    A struct to describe some observation data that we want to compare against.
+
+"""
+struct ObsDataSource
+
+    # This struct is not concretely typed, but we don't care about performance here. We only
+    # care about beauty.
+
+    """Path of the NetCDF file"""
+    path::AbstractString
+
+    """Name of the variable of interest in the NetCDF file"""
+    var_name::AbstractString
+
+    """Name of the time dimension in the NetCDF file"""
+    time_name::AbstractString
+    """Name of the longitude dimension in the NetCDF file"""
+    lon_name::AbstractString
+    """Name of the latitude dimension in the NetCDF file"""
+    lat_name::AbstractString
+
+    """Factor that has to be multiplied to the data to have the same units as our
+    simulations"""
+    unit_conversion_factor::Real
+
+    """The NCDataset associated to the file"""
+    ncdataset::NCDatasets.NCDataset
+end
+
+function ObsDataSource(;
+    path,
+    var_name,
+    time_name = "time",
+    lon_name = "lon",
+    lat_name = "lat",
+    unit_conversion_factor = 1,
+)
+
+    ncdataset = NCDatasets.NCDataset(path)
+
+    return ObsDataSource(
+        path,
+        var_name,
+        time_name,
+        lon_name,
+        lat_name,
+        unit_conversion_factor,
+        ncdataset,
+    )
+end
+
+"""
+    Base.close(ds::ObsDataSource)
+
+Close the file associated to `ds`.
+"""
+function Base.close(ds::ObsDataSource)
+    NCDatasets.close(ds.ncdataset)
+end
+
+"""
+
+    A struct to describe some simulation data.
+
+"""
+struct SimDataSource
+
+    # This struct is not concretely typed, but we don't care about performance here. We only
+    # care about beauty.
+
+    """Path of the simulation output"""
+    path::AbstractString
+
+    """Short name of the variable of interest"""
+    short_name::AbstractString
+
+    """Reduction to consider"""
+    reduction::AbstractString
+
+    """Period of the reduction (e.g., 1d, 30d)"""
+    period::AbstractString
+
+    """ClimaAnalysis OutputVar"""
+    var::ClimaAnalysis.OutputVar
+
+    """Simulation longitudes and latitudes"""
+    lonlat::Tuple{AbstractArray, AbstractArray}
+end
+
+function SimDataSource(;
+    path,
+    short_name,
+    reduction = "average",
+    period = "10d",
+)
+
+    sim = ClimaAnalysis.SimDir(path)
+    var = get(sim; short_name, reduction, period)
+
+    lonlat = (var.dims["lon"], var.dims["lat"])
+
+    return SimDataSource(path, short_name, reduction, period, var, lonlat)
+end
+
+function data_at_date(sim_ds::SimDataSource, target_date::Dates.DateTime)
+    start_date = Dates.DateTime(sim_ds.var.attributes["start_date"])
+    time_diff_seconds = Dates.value(target_date - start_date) / 1e3
```

Comment:
Why is it divided by 1e3?

Reply:
Dates.value is in milliseconds, but I changed to make it more readable.